File: J9Ch/src/S_6_Dynamic_Problem_II/Optional_11/_125_Backpack_II_Medium.cpp

class Solution {
public:
    /**
     * @param m: An integer m denotes the size of a backpack
     * @param A & V: Given n items with size A[i] and value V[i]
     */
    int backPackII(int m, vector<int> A, vector<int> V) {
        // write your code here
        int f[m + 1];
        for (int i = 0; i <=m ; ++i) f[i] = 0;
        int n = A.size() , i, j;
        for(i = 0; i < n; i++){
            for(j = m; j >= A[i]; j--){
                f[j] = max(f[j], f[j - A[i]] + V[i]);
            }
        }
        return f[m];
    }
};

File: J9Ch/src/S_6_DP_II/Optional_11/python_cpp/_125_Backpack_II_Medium.py

class Solution:
    # @param m: An integer m denotes the size of a backpack
    # @param A & V: Given n items with size A[i] and value V[i]
    def backPackII(self, m, A, V):
        # write your code here
        f = [0 for i in xrange(m+1)]
        n = len(A)
        for i in range(n):
            for j in xrange(m, A[i]-1, -1):
                f[j] = max(f[j] , f[j-A[i]] + V[i])
        return f[m]

